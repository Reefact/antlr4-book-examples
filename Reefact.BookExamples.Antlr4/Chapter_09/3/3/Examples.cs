#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Reefact.BookExamples.Antlr4.Chapter_09._1;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_09._3._3 {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class Examples {

        [Fact]
        public void single_token_deletion_mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("class T {{ int i; }");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void single_token_deletion_output() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("class T {{ int i; }");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string output = grun.GetOutput();
            // Verify
            Approvals.Verify(output);
        }

    }

}