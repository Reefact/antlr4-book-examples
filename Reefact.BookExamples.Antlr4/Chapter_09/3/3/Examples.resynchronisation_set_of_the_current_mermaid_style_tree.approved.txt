graph TD
	1["prog"] --> 2["classDef"]
	2 --> 3["class"]
	2 --> 4["T"]
	2 --> 5["{"]
	2 --> 6["member"]
	6 --> 7["int"]
	6 --> 8["x"]
	6 --> 9[";"]
	2 --> 10[";"]:::error
	2 --> 11["#0060;missing '}'#0062;"]:::error
	1 --> 12["classDef"]
	12 --> 13["class"]
	12 --> 14["U"]
	12 --> 15["{"]
	12 --> 16["member"]
	16 --> 17["int"]
	16 --> 18["y"]
	16 --> 19[";"]
	12 --> 20["}"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;